core/src/main/scala/io/finch/Module.scala

```diff
@@ -0,0 +1,15 @@
+package io.finch
+
+import cats.effect.Effect
+import io.finch.syntax.EndpointMappers
+
+abstract class Module[F[_]](implicit val E: Effect[F]) extends Endpoints[F]
```

Comment:
I'm thinking. Maybe we don't have to capture `implicit val E: Effect[F]` here? Monix's builders work this way - only `F[_]` needs to be captured, which means we can resolve the implicit later.

What's interesting is this would allow us to downgrade `Module` to a trait, which gives users the ability to mix it into any existing class should they need to.

This is, obviously, something we can address later, when we move this work under a separate branch.

Reply:
In terms of the performance it should be better. `Effect[F]` is captured once even if it's resolved from `def`